enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,205 @@
+
+
+### Workflow Description
+- The user will enable the AutomatedBackupNoConfig feature gate.
+- A `EtcdBackup` CR is being installed on the cluster.
+- A PVC and PV are being created to save the backup data reliably.
+
+
+### API Extensions
+- No API changes are required. 
+- See https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions
+
+### Topology Considerations
+TBD
+#### Hypershift / Hosted Control Planes
+TBD
+#### Standalone Clusters
+TBD
+#### Single-node Deployments or MicroShift
+TBD
+
+### Implementation Details/Notes/Constraints [optional]
+- Need to agree on a default schedule.
+- Need to agree on a default retention policy.
+
+- Several options exist for the default `PVCName`.
+  - Relying on `dynamic provisioning` is sufficient, however not an option for `SNO` or `BM` clusters.
+  - Utilising `local storage operator` is a proper solution, however installing a whole operator is too much overhead.
+  - The most viable solution to cover all OCP variants is to use `local volume`.
+    Please find below this solution's Pros & Cons.
+    - Pros :- 
+      - The PV will be bound to a specific master node.
+      - The Backup Pod will be scheduled always to this master node since it uses a PVC bound to this PV.
+      - Using an SSD is possible, since local volume allows mounting it into a Pod.
+      - Having the Backup Pod scheduling deterministic is vital, since the retention policy needs the recent backup in place to work.
+    - Cons :-
+      - If the master node where the PV is mounted became unhealthy/unavailable/unreachable etc.
+      - The backups are no longer accessible, also taking new backups is no longer possible.
+      - Ideally, the backups should be taken on a round-robin fashion on each master node to avoid overwhelming a specific etcd member more than others.
+      - Also spreading the backups among all etcd cluster members provide guarantees for disaster recovery in case of losing two members at the same time.
+      - However, using the `local volume` option over all the master nodes will be complicated and error prune.
+
+### Risks and Mitigations
+
+When the backups are configured to be saved to a `local` type PV, the backups are all saved to a single master node where the PV is provisioned on the local disk.
+
+In the event of the node becoming inaccessible or unscheduled, the recurring backups would not be scheduled. The periodic backup config would have to be recreated or updated with a different PVC that allows for a new PV to be provisioned on a node that is healthy.
+
+If we were to use `localVolume`, we need to find a solution for balancing the backups across the master nodes, some ideas could be
+
+- create a PV for each master node using `localVolume` and the backup controller from CEO should take care of balancing the backups across the `available` and `healthy` volumes.
```

Comment:
would a DaemonSet do that for us?

Reply:
Good point. 

- We can use DS and use node affinity to run the DS pods only on master nodes. However, I am not aware of how the storage will be handled in this case. 

- Is there is a way to keep the backups in round-robin fashion among all master nodes ? 

- Also what about STS ? wdyt ?

Reply:
yep, STS could also work, especially with the storage. Would also be useful to have this compared to the static pod/hostpath approach

Reply:
Alright, I think if we have STS, then the Pods, PVC, and PVs are being managed together

I think we can in this case create backups on RoundRobin manner across the master nodes. 

For BM, I think we can still use local volume as storage option and being managed by STS

Reply:
How does this round robin backup work?

Reply:
I read the original enhancement - https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md and I understand this now better. But still this begs the question - how do you know which snapshot to restore from? What if you take snapshot of a etcd replica which was behind and is just catching up with other master nodes? Is that possible?

Reply:
So we will make sure that the backup is taken from the a member whose log is identical to the `leader`. This way we make sure that we are not lagging behind.

But, could this approach work ?

Reply:
Why do you want to handle the balancing? That's not what you should care for at all.

Reply:
Yes I agree, therefore the `SideCar` approach is the most appropriate